yb-voyager/src/query/queryissue/detectors_test.go

```diff
@@ -141,27 +143,26 @@ func TestColumnRefDetector(t *testing.T) {
         HAVING COUNT(*) > 1;`,
 	}
 
-	detector := NewColumnRefDetector()
 	for _, sql := range systemColumnSqls {
+		detector := NewColumnRefDetector(sql)
 		parseResult, err := queryparser.Parse(sql)
 		assert.NoError(t, err, "Failed to parse SQL: %s", sql)
 
 		visited := make(map[protoreflect.Message]bool)
-		unsupportedConstructs := []string{}
 
 		processor := func(msg protoreflect.Message) error {
-			constructs, err := detector.Detect(msg)
+			err := detector.Detect(msg)
 			if err != nil {
 				return err
 			}
-			unsupportedConstructs = append(unsupportedConstructs, constructs...)
 			return nil
 		}
 
 		parseTreeMsg := queryparser.GetProtoMessageFromParseTree(parseResult)
 		err = queryparser.TraverseParseTree(parseTreeMsg, visited, processor)
 		assert.NoError(t, err)
-		assert.Contains(t, unsupportedConstructs, SYSTEM_COLUMNS_NAME, "System Columns not detected in SQL: %s", sql)
+		assert.Equal(t, 1, len(detector.GetIssues()), "Expected 1 issue for SQL: %s", sql)
```

Comment:
nit: `issues := detector.GetIssues()` and use issues further no need to call GetIssues() twice